modules/ticket_selector/ProcessTicketSelectorPostData.php

```diff
@@ -192,12 +175,19 @@ protected function processQuantity($input_key)
                 )
             );
         }
-        ksort($row_qty);
-        // cycle thru values
-        foreach ($row_qty as $qty) {
-            $qty = absint($qty);
-            // sanitize as integers
-            $this->valid_data[ self::DATA_KEY_QUANTITY ][]     = $qty;
+        $max_atndz = $this->valid_data[ self::DATA_KEY_MAX_ATNDZ ];
+        // if max attendees is 1 then the incoming row qty array
+        // will only have one element and the value will be the ticket ID
+        // ex: row qty = [ 0 => TKT_ID ]
+        if ($max_atndz === 1 && count($row_qty) === 1) {
+            // so we'll grab that ticket ID, use it as the key, and set the value to 1
+            // ex: row qty = [ TKT_ID => 1 ]
+            $ticket_id = array_pop($row_qty);
+            $row_qty = [ $ticket_id => 1 ];
+        }
+        foreach ($this->valid_data[ self::DATA_KEY_TICKET_ID ] as $ticket_id) {
+            $qty = isset($row_qty[ $ticket_id ]) ? $row_qty[ $ticket_id ] : 0;
+            $this->valid_data[ self::DATA_KEY_QUANTITY ][ $ticket_id ]     = $qty;
```

Comment:
this means that regardless of the TS format (dropdown, radio buttons, DWMTS) the final valid data array will have the ticket selections in the format: `[ TKT ID => QTY ]`

Reply:
Liking this! Much easier to follow 👍